### PR TITLE
Chapter 12 - Adding Animations

### PR DESCRIPTION
**Pull request by ViolaRafael**
Adds animations in the change of login to sign in,
Adds animations to order details,
Adds animations to page transitions.

### CODE REVIEW

File: lib/components/auth_form.dart [hunk 1]
```diff
@@ -22,12 +23,60 @@ class _AuthFormState extends State<AuthForm> {
     'password': '',
   };
 
+  AnimationController? _controller;
+  Animation<double>? _opacityAnimation;
+  Animation<Offset>? _slideAnimation;
+
   bool _isLogin() => _authMode == AuthMode.login;
-  bool _isSignUp() => _authMode == AuthMode.signup;
+  // bool _isSignUp() => _authMode == AuthMode.signup;
```
Comment (samuelpedropma) [inline lib/components/auth_form.dart]: Remove comment if isn't necessary

File: lib/components/auth_form.dart [hunk 2]
```diff
@@ -88,11 +137,14 @@ class _AuthFormState extends State<AuthForm> {
...
       shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(10)),
-      child: Container(
-        padding: const EdgeInsets.all(16),
-        height: _isLogin() ? 310 : 400,
-        width: deviceSize.width * 0.75,
-        child: Form(
+      child: AnimatedContainer(
+          padding: const EdgeInsets.all(16),
+          height: _isLogin() ? 310 : 400,
+          //height: _heightAnimation?.value.height ?? (_isLogin() ? 310 : 400),
```
Comment (samuelpedropma) [inline lib/components/auth_form.dart]: Remove if isn't necessary

File: lib/components/auth_form.dart [hunk 3]
```diff
@@ -122,21 +174,34 @@ class _AuthFormState extends State<AuthForm> {
...
-                      labelText: 'Confirmar Palavra-passe'),
-                  keyboardType: TextInputType.emailAddress,
-                  obscureText: true,
-                  validator: _isLogin()
-                      ? null
-                      : (_password) {
+                AnimatedContainer(
+                  constraints: BoxConstraints(
+                    minHeight: _isLogin()? 0: 60,
+                    maxHeight: _isLogin()? 0 : 120,
```
Comment (samuelpedropma) [inline lib/components/auth_form.dart]: Leave spaces between the ternary operator and the values

**Review by samuelpedropma — CHANGES_REQUESTED** [review]

File: lib/components/auth_form.dart (hunk 1 above)
Comment (ViolaRafael) [reply lib/components/auth_form.dart]: removed

File: lib/components/auth_form.dart (hunk 2 above)
Comment (ViolaRafael) [reply lib/components/auth_form.dart]: removed

File: lib/components/auth_form.dart (hunk 3 above)
Comment (ViolaRafael) [reply lib/components/auth_form.dart]: fixed

File: lib/components/auth_form.dart [hunk 4]
```diff
@@ -176,7 +174,7 @@ class _AuthFormState extends State<AuthForm>
               ),
                 AnimatedContainer(
                   constraints: BoxConstraints(
-                    minHeight: _isLogin()? 0: 60,
+                    minHeight: _isLogin()? 0 : 60,
```
Comment (samuelpedropma) [inline lib/components/auth_form.dart]: Format document

**Review by samuelpedropma — CHANGES_REQUESTED** [review]

**Review by samuelpedropma — APPROVED** [review]

**Pull request merged**